lms/djangoapps/teams/csv.py

```diff
@@ -19,3 +28,217 @@ def load_team_membership_csv(course, response):
+        # the currently selected user
+        self.user = ''
+        self.number_of_record_added = 0
+        # stores the course module that will be used to get course metadata
+        self.course_module = ''
+        # stores the course for which we are populating teams
+        self.course = course
+        self.max_erros = 0
+        self.max_erros_found = False
+
+    @property
+    def import_succeeded(self):
+        """
+        Helper wrapper that tells us the status of the import
+        """
+        return len(self.validation_errors) == 0
+
+    def set_team_membership_from_csv(self, input_file):
+        """
+        Assigns team membership based on the content of an uploaded CSV file.
+        Returns true if there were no issues.
+
+        Arguments:
+            course (CourseDescriptor): Course module for which team membership needs to be set.
+        """
+        self.course_module = modulestore().get_course(self.course.id)
+        all_rows = [row for row in csv.reader(input_file.read().decode('utf-8').splitlines())]
+        header_row = all_rows[0]
+        if self.validate_teamsets(header_row):
+            # process student rows:
+            for i in range(1, len(all_rows)):
+                row_data = all_rows[i]
+                if row_data[0]:  # avoid processing rows with empty user names (excel copy and paste)
+                    user = self.reset_user(row_data[0])
+                    if user is None:
+                        continue
+                    if self.validate_user_entry(user) is False:
+                        row_data[0] = None
+                        continue
+
+                    row_data[0] = user
+
+                    if self.validate_user_to_team(row_data) is False:
+                        return False
+
+            if not self.validation_errors:
+                for i in range(1, len(all_rows)):
+                    self.add_user_to_team(all_rows[i])
+                return True
+            else:
+                return False
+        return False
+
+    def validate_teamsets(self, header_row):
+        """
+        Validates team set names. Returns true if there are no errors.
+        Also populates teh teamset_names_list.
+        header_row is the list representation of the header row of the input file. It will have
+        the following format:
+        user, mode, <teamset_1_name>,...,<teamset_n_name>
+        where teamset_X_name must be a valid name of an existing teamset.
+        """
+        for i in range(2, len(header_row)):
+            team_config = self.course_module.teams_configuration
+            if not header_row[i] in [ts.teamset_id for ts in team_config.teamsets]:
+                self.validation_errors.append("Teamset named " + header_row[i] + " does not exist.")
+                return False
+            self.teamset_names.append(header_row[i])
+            self.user_ids_by_teamset_index[i] = {m.user_id for m in CourseTeamMembership.objects.filter
```

Comment:
I think the user-provided CSV is going to refer to teamset/topic names, not ids, so you'll probably have to do some mapping of teamset ids to teamset names.  Check out https://github.com/edx/edx-platform/blob/master/openedx/core/lib/teams_config.py

Reply:
Confirming with PM.